realm/realm-library/proguard-rules.pro

```diff
@@ -4,3 +4,5 @@
 -keep @io.realm.internal.Keep class * { *; }
 -dontwarn javax.**
 -dontwarn io.realm.**
+-keep class io.realm.RealmCollection
```

Comment:
Those interfaces were removed when ProGuard shrinks code, you can check it by using `-printusage`.

Yeah this is mainly because of GSON can't create the Type correctly.